packages/vite-plugin-svelte/src/utils/options.ts

```diff
@@ -199,14 +200,23 @@ export function buildExtraViteConfig(
 
 	if (configEnv.command === 'serve') {
 		extraViteConfig.optimizeDeps = buildOptimizeDepsForSvelte(
-			svelteDeps.filter((dep) => dep.type === 'component-library'),
+			svelteComponentLibDeps,
 			options,
 			config.optimizeDeps
 		);
 	}
 
+	// for ssr build, we include all svelte deps to resolve `svelte/ssr`.
+	// for other cases, we can ignore `svelte/ssr` as it would make development faster
+	// and also because vite doesn't handle them properly.
+	// see https://github.com/sveltejs/vite-plugin-svelte/issues/168
+	// see https://github.com/vitejs/vite/issues/2579
 	// @ts-ignore
-	extraViteConfig.ssr = buildSSROptionsForSvelte(svelteDeps, options, config);
+	extraViteConfig.ssr = buildSSROptionsForSvelte(
+		options.isBuild && config.build?.ssr ? svelteDeps : svelteComponentLibDeps,
```

Comment:
should this be moved into the build function? it gets the options passed in and also checks isBuild  and SSR internally.

the way it is now, its a bit harder to grasp what's added exactly BC the input varies